### ISSUE TITLE
ENH/CLN: Refactor API

### ISSUE THREAD

**Issue by NickleDave** [vocalpy/vak#663]
Currently it's not clear what the "public" API for vak is.

Modules that seem like they should be at the top-level and easy to access by a user are hidden in a `core` package. In terms of the repo structure, some (all?) of these modules will need to turn into sub-packages as we add different model families. I.e, the current `train` function that has specifics related to frame classification models will need to be a higher-level dispatch function that calls `train.frame_classification_model`, `train.object_detection_model`, etc.

This is most crucial right now for the `prep` module, where we're going to need to add different functions for different dataset types, e.g. `prep.spectrogram_dataset`, `prep.audio_dataset`, `prep.segment_dataset`, etc.

In contrast to functions that seem like they should be top-level but are hidden in `core`, there's a lot of modules at the top-level that contain what are basically helper and utility functions used by modules in core. Most obviously, `vak.io` is only used by `vak.core.prep.prep`. Similarly `vak.annotation` is used by `prep` as well as the dataset classes.

I would prefer to move these helper and utility functions down a level so they don't seem like a "public" API, and then move the `core` packages up a level.

I want to do this now rather than later because I think it will only become more painful to deal with the API as-is when we are adding new models, datasets, etc.

- [ ] Move functions from `core` up to top-level: eval, learncurve, predict, prep, train. Go ahead and make these all sub-packages where we import all from a module in the sub-package, to avoid having to type `vak.train.train.train`
- [ ] probably just go ahead and rename the current `prep` to `prep.windowed_frame_classification_dataset` since all the function parameters will be specific to that
- [ ] move `io` into `prep` and rename functions appropriately 
- [ ] add `common` sub-package for functions shared across prep / train / learncurve / etc?
  - Basically a `util` but at least the intent is a little more clear. I know I'm grabbing some functions out of `prep` that I'm using in train/learncurve/etc like `validate_and_get_timebin_dur`
